mlflow/models/model.py

```diff
@@ -688,16 +699,33 @@ def log(
             # We check signature presence here as some flavors have a default signature as a
             # fallback when not provided by user, which is set during flavor's save_model() call.
             if mlflow_model.signature is None and (
-                tracking_uri == "databricks" or get_uri_scheme(tracking_uri) == "databricks"
+                tracking_uri == "databricks"
+                or get_uri_scheme(tracking_uri) == "databricks"
             ):
                 _logger.warning(_LOG_MODEL_MISSING_SIGNATURE_WARNING)
-            mlflow.tracking.fluent.log_artifacts(local_path, mlflow_model.artifact_path, run_id)
+            mlflow.tracking.fluent.log_artifacts(
+                local_path, mlflow_model.artifact_path, run_id
+            )
+
+            # if the model_config kwarg is passed in, then log the model config as an params
+            try:
+                if "model_config" in kwargs:
+                    model_config = kwargs["model_config"]
+                    if isinstance(model_config, str):
```

Comment:
Wrapping the entire statement in a try / catch might make debugging a little hard to figure out. 
If I submit a string that points to a non-existent YAML config, I would get a file access error, but if there is an unserializable object within the configuration that cannot be written as a param entry, it might be a bit hard to figure out what is going wrong. 
I'd recommend splitting up the exception handling to cover the two primary failure modes